### PR TITLE
Fix/saml err output

### LINKED ISSUE #1652: SAML: Not getting error output when running inside PF process
Seems the output of the Lasso errors is lost when running inside a PacketFence process

Using a pure perl one line gives the output. Not within pfcmd.

In the example below, the idp.xml file had a bad xml header.

```
[maml]
description=SAML source
idp_ca_cert_path=/usr/local/pf/conf/ssl/idp-ca.crt
idp_entity_id=https://idp.testshib.org/idp/shibboleth
idp_metadata_path=/usr/local/pf/conf/idp.xml
username_attribute=urn:oid:0.9.2342.19200300.100.1.1
dynamic_routing_module=AuthModule
idp_cert_path=/usr/local/pf/conf/ssl/idp.crt
sp_entity_id=pf-julien.inverse.ca
type=SAML
authorization_source_id=inverse
sp_cert_path=/usr/local/pf/conf/ssl/server.crt
sp_key_path=/usr/local/pf/conf/ssl/server.key
```

Then the difference between both commands:

```
# perl -Ilib -Mpf::authentication -e 'print getAuthenticationSource("maml")->sso_url'`

(process:12545): Lasso-CRITICAL **: libxml2: Start tag expected, '<' not found\n

(process:12545): Lasso-WARNING **: 2016-08-11 12:59:26  Cannot load metadata from /usr/local/pf/conf/idp.xml
Can't create Single-Sign-On URL : Failed to add new provider.

# bin/pfcmd saml testsource maml
You should see the SSO URL below, otherwise, there are errors in your configuration:
Can't create Single-Sign-On URL. Check server side logs for details and validate the SAML configuration. at /usr/local/pf/lib/pf/Authentication/Source/SAMLSource.pm line 158.
```

### PR DESCRIPTION
**Pull request by julsemaan**
# This is not ready and contains a hack to bypass a bug with mod_perl2 and liblasso.
# Description

Make liblasso calls go through a CGI webserver hosted in the webservices
Various ameliorations around error handling in SAML
# Impacts

SAML authentication
# Issue

fixes #1652 
# Delete branch after merge

YES
# NEWS file entries
## Bug Fixes
- Fixed issue with SAML running on recent versions of mod_perl2

### CODE REVIEW

**Comment by louismunro** [comment]
Status please?

**Comment by julsemaan** [comment]
Its an ugly hack to bypass an issue with Lasso.

Unless we find the underlying issue with Lasso, we'll have to stick with this fix

**Comment by louismunro** [comment]
This makes me feel dirty just reading the title.

File: lib/pf/web/saml/assertion.cgi [hunk 1]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/perl
```
Comment (julsemaan) [inline lib/pf/web/saml/assertion.cgi:1]: Need to add executable bit to that file in packaging

**Comment by dwlfrth** [comment]
Check for httpd template configuration. There is an include between aaa and webservices and the new VirtualHost is being instanciated twice.

**Comment by julsemaan** [comment]
fixed

**Comment by julsemaan** [comment]
Need to add this to httpd.aaa.tt.example:
```
Listen 127.0.0.1:9091
<VirtualHost 127.0.0.1:9091 >
  ScriptAlias / [% install_dir %]/lib/pf/web/saml/
  DocumentRoot [% install_dir %]/lib/pf/web/saml/
</VirtualHost>

LoadModule cgi_module modules/mod_cgi.so
LoadModule alias_module modules/mod_alias.so

<Directory [% install_dir%]/lib/pf/web/saml>
  [% IF apache_version == "2.4" %]
  Require all granted
  [% ELSE %]
  Allow from all
  [% END %]
</Directory>

```

**Comment by extrafu** [comment]
I think we can close this, can't we?

**Comment by julsemaan** [comment]
We sure can

**Pull request closed without merging**